### ISSUE TITLE
An error occurred (UnrecognizedClientException) when calling the PutIntegration operation: The security token included in the request is invalid

### ISSUE THREAD

**Issue by avengerpenguin** [Miserlou/Zappa#198]
I get this error randomly part-way through the "Creating API Gateway routes (this only happens once)" stage.

I'd tried a few things and it seems to fail somewhere between 8% and 11% through, which suggests to me that it's not doing the same operation. In fact, on one run it failed on "PutMethodResponse" and not "PutIntegration"

The only thing I can conclude is there is some kind of rate-limiting I'm hitting that Zappa is seeing as key issue, but key pair worked fine for the first 8-11% of the operation.

I can't see a way to "resume" the API Gateway part as it's part of the first-time "zappa deploy" command. That would give me a workaround here at least.

**Comment by avengerpenguin** [comment]
I found a workaround by limiting the `parameter_depth` and `http_methods` config values to precisely what I need (thus limiting the number of put actions on the API Gateway api). The progress bar moved much more quickly and finished before any error could occur.

**Comment by avengerpenguin** [comment]
Of course, there's still a bug or weird behaviour here if I had wanted to use deeper URLs and more methods.

**Comment by Miserlou** [comment]
I haven't encountered this, and it looks like a bit of a bastard to reproduce, but leaving this ticket open for now.

API Gateway path creation needs a real overhaul any, I'd much rather do it using something like troposphere and CloudFormation if possible, doing it one API call at a time is awful. API Gateway didn't used to be available in CF, I don't know if that's still the case.

**Comment by avengerpenguin** [comment]
I assume the code has changed enough now that I could have a check again to see if this still happens?

**Comment by Miserlou** [comment]
I can look into this if you have a way to reproduce, otherwise I can't really do anything about it. We are using CF, so this may not even be relevant. Checks would help though..
